### DISCUSSION TITLE
CSS is not being generated for some given utilities

### DISCUSSION THREAD

**Discussion by Nefcanto** [tailwindlabs/tailwindcss#7576] [Help]
Hi

I read your template and I knew that in this case, it won't help. Thus please let me explain my problem this way:

Here's a picture of one of my components:

![imageedit_0_6780571386](https://user-images.githubusercontent.com/12782347/155060095-564a5ff0-830d-4074-b1e2-5fe06ce4da34.png)

As you can see in this image, `w-full` is working, and `w-10` has also been generated. But for some reason, `h-full` is not being picked up.

I have tried to create a simple separate example and in all examples `h-full` works. However, in a real-world application that is truly complex (using react + material UI +, etc. in more than 200 components), it does not work.

Since my sandbox creation attempt failed for many attempts, this means that Tailwind's engine can't pick `h-full` in the context of this app. So the engine works fine, but in some cases, it can't pick class names. What cases? I can't tell and I can't ask you to look at my repository as you have mentioned duly. 

What I ask for is to help us debug Tailwind on our own side. I tried to find my options for debugging, but there was not much. I especially read [Content Configuration](https://tailwindcss.com/docs/content-configuration) and still could not solve my problem. I'm using the class names in full, I am specific about my content paths (using your own defaults only), 

I built my CSS using `npx tailwindcss -o build.css` as explained [here](https://tailwindcss.com/docs/optimizing-for-production), and I can verify that `h-full` is not generated in the output file.

But how can I know **why**?

My suggestions:

- Let us specify one single file for generation in `npx` and give us the list of all classes you have found in it.
- Create documentation on how to debug these cases when a utility is not being generated
- Somehow teach us through docs about the mechanism with which Tailwind picks class names (for example regular expressions used and how can we extend them and what pitfalls might exist that mess with them, etc.)
- Craft a list of common mistakes that you know the community has done over the years

In the end, I should thank you so much for Tailwind. I have been using Tailwind for like 1 year now and I would choose Tailwind a thousand times more if I go back.

**Comment by adamwathan** [comment]
Hey! Sorry we really can’t help without a reproduction. Can you at a minimum share the actual source code for that component, the full path where that file is located, and your complete Tailwind config file?

Or if you can invite me to a private repo for the project I can also look at that.

**Comment by Nefcanto** [comment]
Thank you @adamwathan for your kind support. We actually rewrite a couple of our components and now the problem is solved. I don't know what might have caused the issue. But I think we need more "debugging tools" for tailwind. By the way, we're all adicted to tailwind and our entire company and sister companies are migrating all of our products to tailwind.

I sincerely thank you for that.
